pubsub/azurepubsub/azurepubsub.go

```diff
@@ -289,7 +289,8 @@ type subscription struct {
 
 // SubscriptionOptions will contain configuration for subscriptions.
 type SubscriptionOptions struct {
-	ListenerTimeout time.Duration
+	ListenerTimeout            time.Duration
+	AckFuncForReceiveAndDelete func() // AckFuncForReceiveAndDelete is used for at-most-once mode (SB Subscription with SubscriptionWithReceiveAndDelete option)
```

Comment:
The comment is long enough that it should go above the field instead of next to it.

It should make it clear that if you set the field, you will get ReceiveAndDelete mode. Something like:

// If nil, the subscription will be in Peek-Lock mode. The Ack method must be called on each message
// to complete it.
// If non-nil, the subscription will be in Receive-and-Delete mode, and this function will be called 
// whenever Ack is called on a message. 
// See the "At-most-once vs. At-least-once Delivery" section in the pubsub package documentation.

Reply:
Ok, I made a small change - in both scenario, the setting MUST match the mode (Nil for Peek-Lock, Not Nil for Receive-and-Delete)